src/app/clusters/basic/basic.cpp

```diff
@@ -54,6 +54,12 @@ class BasicAttrAccess : public AttributeAccessInterface
 
 BasicAttrAccess gAttrAccess;
 
+CHIP_ERROR EncodeStringOnSuccess(CHIP_ERROR status, AttributeValueEncoder & encoder, const char * buf, size_t maxBufSize)
```

Comment:
nit: Make this a static function

Reply:
It is in the anonymous namespace, it is thus static